src/ZoneServer/World/Actors/Characters/Components/CollectionComponent.cs

```diff
@@ -0,0 +1,162 @@
+			: base(character)
+		{
+		}
+
+
+		/// <summary>
+		/// Adds a new empty Collection.
+		/// <param name="collectionId"></param>
+		public bool Add(int collectionId)
+		{
+			lock (_syncLock)
+			{
+				if (ZoneServer.Instance.Data.CollectionDb.TryFind(collectionId, out var collectionData) && !_collectionList.ContainsKey(collectionId))
+				{
+					_collectionList.Add(collectionId, collectionData);
+					_collectionProgress.Add(collectionId, new List<int>());
+					return true;
+				}
+				return false;
+			}
+		}
+
+
+		/// <summary>
+		/// Adds a Collection without informing the client.
+		/// </summary>
+		/// <remarks>
+		/// This is primarily used while the character and its quests are
+		/// loaded from the database.
+		/// </remarks>
+		/// <param name="collectionId"></param>
+		/// <param name="registeredItems"></param>
+		public void AddSilent(int collectionId, List<int> registeredItems)
+		{
+			lock (_syncLock)
+			{
+				if (ZoneServer.Instance.Data.CollectionDb.TryFind(collectionId, out var collectionData))
+				{
+					_collectionList.Add(collectionId, collectionData);
+					_collectionProgress.Add(collectionId, registeredItems);
+					if (registeredItems.Count == collectionData.RequiredItems.Count)
+					{
+						// TODO: Add the reward to the list of earned rewards
+					}
+				}
+			}
+		}
+
+		/// <summary>
+		/// Gets a collection and returns the list of items registered to it.
+		/// </summary>
+		/// <param name="collectionId"></param>
+		/// <param name="registeredItems"></param>
+		/// <returns></returns>
+		public bool TryGet(int collectionId, out List<int> registeredItems)
+		{
+			lock (_syncLock)
+				return _collectionProgress.TryGetValue(collectionId, out registeredItems);
+		}
+
+		/// <summary>
+		/// Returns the total number of registered collections
+		/// </summary>
+		/// <returns></returns>
+		public int Count()
+		{
+			lock (_syncLock)
+				return _collectionList.Count;
+		}
+
+		/// <summary>
+		/// Returns the list of all collections the user has registered
+		/// </summary>
+		/// <returns></returns>
+		public List<int> GetList()
+		{
+			lock (_syncLock)
+				return _collectionList.Keys.ToList();
+		}
+
+		/// <summary>
+		/// Returns the list of items currently registered to the given collection
+		/// This list might be empty if no items are registered
+		/// Returns null if the collection isn't registered at all
+		/// </summary>
+		/// <returns></returns>
+		public List<int> GetProgress(int collectionId)
+		{
+			lock (_syncLock)
+			{ 
+				if (_collectionProgress.TryGetValue(collectionId, out var progress))
+					return progress;
+				return null;
+			}
+		}
+
+
+		/// <summary>
+		/// Registers an item to this collection
+		/// <param name="collectionId"></param>
+		/// <param name="item"></param>
+		/// </summary>
+		public bool RegisterItem(int collectionId, Item item)
+		{
+			lock (_syncLock)
+			{
+				if (_collectionList.TryGetValue(collectionId, out var collectionData))
```

Comment:
The nested ifs in this area are a bit difficult to read. It would arguably be clearer if you employed a structure where you check for invalid states first and put the success at the end.

Instead of this:
```text
if (a)
{
	if (b)
	{
		if (c)
		{
			if (d)
			{
				return true;
			}
		}
	}
}

return false;
```
You would do this:
```text

if (!a)
	return false;

if (!b)
	return false;

if (!c)
	return false;

if (!d)
	return false;

return true;
```

This makes it much easier to read, understand, and maintain. Also, you should not omit the curly braces for an `if` if the conditional code is longer than one line, as per our coding conventions.